include/dts-functions.sh

```diff
@@ -1387,3 +1387,90 @@ show_menu() {
     echo -ne "${RED}${SSH_OPT_UP}${NORMAL} to launch SSH server  ${NORMAL}"
   fi
 }
+
+check_if_fused() {
+file_path="/sys/class/mei/mei0/fw_status"
+
+if [[ ! -f $file_path ]]; then
+    echo "File not found: $file_path"
+    return 2
+fi
+
+hfsts6_value=""
+line_number=1
+while IFS= read -r line; do
+    if [[ $line_number -eq 6 ]]; then
+        hfsts6_value=$line
+        break
+    fi
+    ((line_number++))
+done < "$file_path"
+
+if [[ -z $hfsts6_value ]]; then
+    echo "Failed to read HFSTS6 value"
+    exit 1
+fi
+
+hfsts6_binary=$(echo "ibase=16; obase=2; $hfsts6_value" | bc)
+
+binary_length=${#hfsts6_binary}
+
+# Add leading zeros
+if [ $binary_length -lt 32 ]; then
+  padding=$((32 - $binary_length))
+  zeros=$(printf "%${padding}s" | tr ' ' "0")
+  hfsts6_binary=$zeros$hfsts6_binary
+fi
+
+bit_30_value=${hfsts6_binary:1:1}
+
+if [ $bit_30_value == 0 ]; then
+  return 1
+else
+  return 0
+fi
+}
+
+check_if_boot_guard_enabled() {
+  # MSR cannot be read
+  if ! rdmsr 0x13a -0; then
```

Comment:
Could you link to some documentation & page/section or explanation?
I'm trying to find those registers in

> Intel® 64 and IA-32 Architectures
Software Developer’s Manual
Volume 4:
Model-Specific Registers

Reply:
@miczyg1 can you help here?

Reply:
@miczyg1 @BeataZdunczyk this seem to be related to Dasharo TrustRoot Assesment

Reply:
You will not find this MSR anywhere. This is BootGuard MSR